tests/integration/cluster-users/sar-test.sh

```diff
@@ -276,30 +371,192 @@ function runTests() {
 	testPermissions "create" "secret" "${CUSTOM_NAMESPACE}" "yes"
 
-	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to delete namespace they created"
-	testPermissions "delete" "namespace" "${CUSTOM_NAMESPACE}" "yes"
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to update Secret in the namespace they created"
+	testPermissions "update" "secret" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to get Secret in the namespace they created"
+	testPermissions "get" "secret" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to delete Secret in the namespace they created"
+	testPermissions "delete" "secret" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to list Roles in the namespace they created"
+	testPermissions "list" "roles.rbac.authorization.k8s.io" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to create Role in the namespace they created"
+	testPermissions "create" "role.rbac.authorization.k8s.io" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to get Role in the namespace they created"
+	testPermissions "get" "role.rbac.authorization.k8s.io" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to update Role in the namespace they created"
+	testPermissions "update" "role.rbac.authorization.k8s.io" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to delete Role in the namespace they created"
+	testPermissions "delete" "role.rbac.authorization.k8s.io" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to list RoleBindings in the namespace they created"
+	testPermissions "list" "rolebindings.rbac.authorization.k8s.io" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to create RoleBinding in the namespace they created"
+	testPermissions "create" "rolebinding.rbac.authorization.k8s.io" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to get RoleBinding in the namespace they created"
+	testPermissions "get" "rolebinding.rbac.authorization.k8s.io" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to update RoleBinding in the namespace they created"
+	testPermissions "update" "rolebinding.rbac.authorization.k8s.io" "${CUSTOM_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to delete RoleBinding in the namespace they created"
+	testPermissions "delete" "rolebinding.rbac.authorization.k8s.io" "${CUSTOM_NAMESPACE}" "yes"
 
 	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to create rolebindings to kyma-developer clusterrole in the namespace they created"
 	createRoleBindingForNamespaceDeveloper
 
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should NOT be able to delete any namespace in the cluster"
```

Comment:
I would suggest rephrasing that to `[..] to delete all namespaces in the cluster`. It may be misleading, as one of next assertions is testing that the `NAMESPACE_ADMIN` actually **CAN** delete some namespaces (specifically namespaces they created)

Reply:
How about: "should NOT be able to delete arbitrary namespaces in the cluster" ?